mod/ArchConsoleManager.cs

```diff
@@ -0,0 +1,189 @@
+            }
+            pauseConsole.GetComponentInChildren<InputField>().onEndEdit.AddListener(OnConsoleEntry);
+            consoleText = pauseConsole.GetComponentInChildren<InputField>();
+            pauseConsoleViewer.SetActive(false);
+
+            StartCoroutine(LoopGreeting());
+        }
+
+        // Shows the appropriate consoles when the game is paused or not
+        private void ShowConsoles(bool showPauseConsole)
+        {
+            pauseConsoleViewer.SetActive(showPauseConsole);
+            gameplayConsole.SetActive(!showPauseConsole);
+        }
+
+        /// <summary>
+        /// Adds a new text entry to the in-game consoles
+        /// </summary>
+        /// <param name="text">The text to add to the consoles</param>
+        /// <param name="skipGameplayConsole">Whether to only show text on the pause console</param>
+        /// <param name="skipMasterList">Whether to not save this text between loops</param>
+        public void AddText(string text, bool skipGameplayConsole = false, bool skipMasterList = false)
+        {
+            if (!skipMasterList) consoleMasterList.Add(text);
+            GameObject textEntry = Instantiate(consoleEntry, pauseConsoleContent.transform);
+            textEntry.GetComponent<Text>().text = text;
+            pauseConsoleEntries.Add(textEntry);
+
+            // We don't need to bother editing the Gameplay Console if this is on
+            if (skipGameplayConsole) return;
+
+            for (int i = 0; i < gameplayConsoleEntries.Count; i++)
+            {
+                if (i == gameplayConsoleEntries.Count - 1)
+                {
+                    gameplayConsoleEntries[i].GetComponent<Text>().text = text;
+                }
+                else
+                {
+                    gameplayConsoleEntries[i].GetComponent<Text>().text = gameplayConsoleEntries[i + 1].GetComponent<Text>().text;
+                }
+            }
+
+            if (!isPaused)
+            {
+                gameplayConsole.SetActive(true);
+            }
+            // Attempts to fix weirdness in console layout when multiple messages are received
+            LayoutRebuilder.ForceRebuildLayoutImmediate(gameplayConsole.GetComponent<RectTransform>());
+            LayoutRebuilder.ForceRebuildLayoutImmediate(pauseConsoleContent.GetComponent<RectTransform>());
+        }
+
+        /// <summary>
+        /// Adds a new text entry to the in-game consoles.
+        /// Identical to Randomizer.Instance.ArchConsoleManager.AddText(text), but implemented for convenience.
+        /// </summary>
+        /// <param name="text">The text to add to the consoles</param>
+        /// <param name="skipGameplayConsole">Whether to only show text on the pause console</param>
+        /// <param name="skipMasterList">Whether to not save this text between loops</param>
+        public static void AddConsoleText(string text, bool skipGameplayConsole = false, bool skipMasterList = false)
```

Comment:
They are used in AddText, so I've left the option open in case we need to use them from an outside class in the future. The second argument is used to make it so that the text only appears on the pause console, and the third argument is used to make the text not save, so it doesn't get reapplied between loops.

Reply:
Sorry, I understand what the arguments are _doing_ inside the implementation, I'm questioning whether we need them.

Reply:
They're not needed within the context of what I do, but they're left available for usage outside of the class if needed. Since this method exists only to save some typing, I don't see why it should be less powerful than the method it already calls.